cookies/id_token_test.go

```diff
@@ -0,0 +1,43 @@
+package cookies
+
+import (
+"net/http"
+"net/http/httptest"
+"testing"
+
+. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestUnitIDToken(t *testing.T) {
+
+	var testDomain = "www.test.com"
+	var testIDToken = "test-id-token"
+
+	Convey("GetIDToken", t, func() {
+		Convey("returns cookie value if value is set", func() {
+			req := httptest.NewRequest("GET", "/", nil)
+			req.AddCookie(&http.Cookie{Name: idCookieKey, Value: testIDToken})
+			cookie, _ := GetIDToken(req)
+			So(cookie, ShouldEqual, testIDToken)
+		})
+
+		Convey("returns error if no cookie is set", func() {
+			req := httptest.NewRequest("GET", "/", nil)
+			_, err := GetIDToken(req)
+			So(err, ShouldNotBeNil)
+		})
+	})
+
+	Convey("SetIDToken sets correct cookie", t, func() {
+		rec := httptest.NewRecorder()
+		SetIDToken(rec, testIDToken, testDomain)
+		cookie := rec.Result().Cookies()[0]
+		So(cookie.Value, ShouldEqual, testIDToken)
```

Comment:
I see that a number of the unit tests have these `So` assertion blocks, where we're asserting on individual properties. Unless I've missed anything, it looks like the only dynamic value here is the `cookie.Value` property. Ultimately this feels like a candidate for refactoring

Maybe we could create an `expectedCookie` with all these properties + the dynamic token value, and then we could remove these lines and do something like 

```
expectedCookie := NewExpectedCookie(testIDToken)
So(receivedCookie, ShouldResemble, expectedCookie)
```

Let me know if you've any thoughts either way though

Reply:
Haha, guilty of a Copy and Paste then Find and Replace on those tests ><, will have a look into that.